src/components/TodoItem.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  deleteTodo: (todoId: number) => void;
+  deleteId: number[];
+  selectedaTodo: number | null | undefined;
+  onSelectedTodo: (id: number | null) => void;
+  editTodo: (todoID: number, updatedTodo: Todo) => void;
+  checkedTodo: (todo: Todo) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  deleteId,
+  selectedaTodo,
+  onSelectedTodo,
+  editTodo,
+  checkedTodo,
+}) => {
+  const deleteItem = deleteId.includes(todo.id);
+  const [renameTodo, setRenameTodo] = useState<string>(todo.title);
+
+  const handleRenameTodo = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRenameTodo(event.target.value);
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    if (renameTodo !== todo.title && editTodo) {
+      editTodo(todo.id, { ...todo, title: renameTodo });
+    }
+
+    if (onSelectedTodo) {
+      onSelectedTodo(null);
+    }
+  };
+
+  const handleKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setRenameTodo(todo.title);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (renameTodo !== todo.title && editTodo) {
+      editTodo(todo.id, { ...todo, title: renameTodo });
+    }
+
+    if (onSelectedTodo) {
+      onSelectedTodo(null);
+    }
+  };
+
+  return (
+    <>
```

Comment:
There is no need to use React Fragment, while you are returning just one element in jsx